packages/atlas/src/views/studio/VideoWorkspace/VideoWorkspace.tsx

```diff
@@ -99,22 +72,94 @@ export const VideoWorkspace: React.FC = React.memo(() => {
-            onThumbnailFileChange={handleThumbnailFileChange}
-            onVideoFileChange={handleVideoFileChange}
-            fee={0}
-            thumbnailHashPromise={thumbnailHashPromise}
-            videoHashPromise={videoHashPromise}
-            setAssetsCache={setAssetsCache}
+          <ScrollContainer actionBarHeight={actionBarHeight}>
+            <VideoForm setFormStatus={setFormStatus} onSubmit={handleSubmit} />
+          </ScrollContainer>
+
+          <VideoWorkspaceActionBar
+            isEdit={isEdit}
+            // form can be submitted if both:
+            // 1. form is valid
+            // 2. the video is a new one OR the form is dirty (some edit has been made)
+            canSubmit={(formStatus?.isValid && (!isEdit || formStatus?.isDirty)) || false}
+            canReset={formStatus?.isDirty || false}
+            onSubmit={formStatus?.triggerFormSubmit}
+            onReset={formStatus?.resetForm}
+            onResize={setActionBarHeight}
           />
         </Container>
       </CSSTransition>
     </>
   )
 })
-
 VideoWorkspace.displayName = 'VideoWorkspace'
+
+type VideoWorkspaceActionBarProps = {
+  isEdit: boolean
+  canReset: boolean
+  canSubmit: boolean
+  onSubmit?: () => void
+  onReset?: () => void
+  onResize?: (height: number) => void
+}
+
+const VideoWorkspaceActionBar: React.FC<VideoWorkspaceActionBarProps> = ({
+  isEdit,
+  canReset,
+  canSubmit,
+  onSubmit,
+  onReset,
+  onResize,
+}) => {
+  const mdMatch = useMediaMatch('md')
+  const [actionBarRef, actionBarBounds] = useMeasure()
+
+  const isActive = !isEdit || canSubmit
+  const height = isActive ? actionBarBounds.height : 0
+
+  // send update to VideoWorkspace whenever height changes
+  useEffect(() => {
+    if (!onResize) {
+      return
+    }
+
+    onResize(height)
+  }, [height, onResize])
+
+  return (
+    <StyledActionBar
+      ref={actionBarRef}
+      isEdit={isEdit}
+      primaryText="Fee: 0 Joy"
+      secondaryText="For the time being no fees are required for blockchain transactions. This will change in the future."
+      primaryButton={{
+        text: isEdit ? 'Publish changes' : 'Upload',
+        disabled: !canSubmit,
+        onClick: onSubmit,
+        tooltip: canSubmit
+          ? undefined
+          : {
+              headerText: 'Fill all required fields to proceed',
+              text: 'Required: video file, thumbnail image, title, category, language',
+            },
+      }}
+      secondaryButton={{
+        visible: canReset,
+        text: 'Cancel',
+        onClick: () => onReset?.(),
+        icon: <SvgControlsCancel width={16} height={16} />,
+      }}
+      draftBadge={{
+        visible: !isEdit,
+        text: mdMatch ? 'Drafts are saved automatically' : 'Saving drafts',
+        tooltip: {
+          text: 'Drafts system can only store video metadata. Selected files (video, thumbnail) will not be saved as part of the draft.',
+        },
+      }}
+      fullWidth={true}
```

Comment:
This `fullWidth` doesn't do anything. This can be safely removed here and in ActionBarProps